### PR TITLE
added function to upload the model artifact to azure blob storage

### LINKED ISSUE #3: Add support to log an artifact during run
During a run we want to upload a model to a container in Azure.

I'm envisioning the structure to look like a container name ModelRepository and each model being named the unique GUID for the run.

### PR DESCRIPTION
**Pull request by AnoojNair**
Fixes #3 

Contains the changes to upload the model artifact to a container in azure blob storage.

1. Container name is 'model-repository'. If it doesn't the method creates it first.
2. The blob name is the runId with a .zip extenion.
3. Doesn't apply any permissions on the blob.

### CODE REVIEW

File: MLOps.NET.Azure/Storage/StorageAccountModelRepository.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MLOps.NET.Storage
+{
+    internal sealed class StorageAccountModelRepository : IModelRepository
+    {
+        private readonly BlobContainerClient blobContainerClient;
+        private const string _containerName = "model-repository";
```
Comment (aslotte) [inline MLOps.NET.Azure/Storage/StorageAccountModelRepository.cs:12]: My bad, I should have added a coding style guide to the repo. Just to keep it consistent I think we will not be using underscore for private member variables (see blobContainerClient above).

File: MLOps.NET/MLLifeCycleManager.cs [hunk 2]
```diff
@@ -29,9 +31,21 @@ public async Task LogMetricAsync(Guid runId, string metricName, double metricVal
             await MetaDataStore.LogMetricAsync(runId, metricName, metricValue);
         }
 
+        /// <summary>
```
Comment (aslotte) [inline MLOps.NET/MLLifeCycleManager.cs:34]: I'm not against summary comments, but we should probably be consistent (meaning I should probably update the other methods too... :) 

With that said, this comment should probably be made more generic as what we are doing is just storing the model in a repository. In the future that can be other places than Azure, e.g. on a local fileshare

Comment (AnoojNair) [reply MLOps.NET/MLLifeCycleManager.cs:34]: Ohh yes,  have updated the comment to be more generic 👍

File: MLOps.NET.Azure/Storage/StorageAccountModelRepository.cs (hunk 1 above)
Comment (AnoojNair) [reply MLOps.NET.Azure/Storage/StorageAccountModelRepository.cs]: Done 👍

**Pull request merged**